src/MarketingPageAcceptanceTestsSpecflow/Steps/ClientApplication/SharedClientApplicationTypesSteps.cs

```diff
@@ -123,6 +123,21 @@ public void GivenASupplierHasSavedAllMandatoryDataOnTheNativeDesktopClientApplic
             GivenASupplierHasSavedAllMandatoryDataOnTheClientApplicationTypeSub_SectionsExceptForX(clientApplicationType, section);
         }
 
+        [Given(@"that (.*) has been completed for (Browser based|Native mobile or tablet|Native desktop)")]
```

Comment:
My gut feel is that this should be a unique step for connection details, but i can see why you're doing it. Happy to leave it as it is for now and re-evaluate as the switch statement grows